src/Wallabag/CoreBundle/Event/Subscriber/SchemaAdapterSubscriber.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+namespace Wallabag\CoreBundle\Event\Subscriber;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\DBAL\Platforms\MySQLPlatform;
+use Doctrine\ORM\Tools\Event\GenerateSchemaEventArgs;
+
+class SchemaAdapterSubscriber implements EventSubscriber
+{
+    private string $databaseTablePrefix;
+
+    public function __construct(string $databaseTablePrefix)
+    {
+        $this->databaseTablePrefix = $databaseTablePrefix;
+    }
+
+    public function getSubscribedEvents()
+    {
+        return ['postGenerateSchema'];
+    }
+
+    public function postGenerateSchema(GenerateSchemaEventArgs $eventArgs)
+    {
+        $platform = $eventArgs->getEntityManager()->getConnection()->getDatabasePlatform();
+
+        if (!$platform instanceof MySQLPlatform) {
+            return;
+        }
+
+        $schema = $eventArgs->getSchema();
+
+        $entryTable = $schema->getTable($this->databaseTablePrefix . 'entry');
+        $entryTable->addOption('collate', 'utf8mb4_unicode_ci');
+        $entryTable->addOption('charset', 'utf8mb4');
+
+        $tagTable = $schema->getTable($this->databaseTablePrefix . 'tag');
+        $tagTable->addOption('collate', 'utf8mb4_bin');
+        $tagTable->addOption('charset', 'utf8mb4');
```

Comment:
Why it's better to have it here instead of inside the entity?

Reply:
because it's specific to MySQL, and here we can set those only for MySQL, without impacting other DBMS.
on entities it's for all DBMS